README.md
![Community-Project](https://gitlab.com/softbutterfly/open-source/open-source-office/-/raw/master/banners/softbutterfly-open-source--banner--community-project.png)

![PyPI - Supported versions](https://img.shields.io/pypi/pyversions/wagtail-sb-socialnetworks)
![PyPI - Package version](https://img.shields.io/pypi/v/wagtail-sb-socialnetworks)
![PyPI - Downloads](https://img.shields.io/pypi/dm/wagtail-sb-socialnetworks)
![PyPI - MIT License](https://img.shields.io/pypi/l/wagtail-sb-socialnetworks)

[![Build Status](https://www.travis-ci.org/softbutterfly/wagtail-sb-socialnetworks.svg?branch=develop)](https://www.travis-ci.org/softbutterfly/wagtail-sb-socialnetworks)
[![Codacy Badge](https://app.codacy.com/project/badge/Grade/3d703e48c1e44e9b830da5026f07c52d)](https://www.codacy.com/gh/softbutterfly/wagtail-sb-socialnetworks/dashboard?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=softbutterfly/wagtail-sb-socialnetworks&amp;utm_campaign=Badge_Grade)
[![Codacy Badge](https://app.codacy.com/project/badge/Coverage/3d703e48c1e44e9b830da5026f07c52d)](https://www.codacy.com/gh/softbutterfly/wagtail-sb-socialnetworks/dashboard?utm_source=github.com&utm_medium=referral&utm_content=softbutterfly/wagtail-sb-socialnetworks&utm_campaign=Badge_Coverage)
[![codecov](https://codecov.io/gh/softbutterfly/wagtail-sb-socialnetworks/branch/master/graph/badge.svg?token=pbqXUUOu1F)](https://codecov.io/gh/softbutterfly/wagtail-sb-socialnetworks)
[![Requirements Status](https://requires.io/github/softbutterfly/wagtail-sb-socialnetworks/requirements.svg?branch=master)](https://requires.io/github/softbutterfly/wagtail-sb-socialnetworks/requirements/?branch=master)

# Wagtail Social Networks

Wagtail package to manage sites social network profiles.

## Requirements

- Python 3.8, 3.9, 3.10

## Install

```bash
pip install wagtail-sb-socialnetworks
```

## Usage

Add `wagtail.contrib.settings` and `wagtail_sb_socialnetworks` to your `INSTALLED_APPS` settings

```
INSTALLED_APPS = [
  # ...
  "wagtail.contrib.settings",
  "wagtail_sb_socialnetworks",
  # ...
]
```

## Docs

- [Ejemplos](https://github.com/softbutterfly/wagtail-sb-socialnetworks/wiki)
- [Wiki](https://github.com/softbutterfly/wagtail-sb-socialnetworks/wiki)

## Changelog

All changes to versions of this library are listed in the [change history](CHANGELOG.md).

## Development

Check out our [contribution guide](CONTRIBUTING.md).

## Contributors

See the list of contributors [here](https://github.com/softbutterfly/wagtail-sb-socialnetworks/graphs/contributors).


CHANGELOG.md
# Changelog

## [Unreleased]

## [0.4.0] - 2022-10-02

* Update dependencies
* Add support for wagtail 4.0
* Add demo project
* Make icon field optional for SocialNetwork model
* Rename sitesettings to site_settings for SiteSocialNetworkProfile model

## [0.3.0] - 2022-06-19

## [0.2.0] - 2022-05-17

* Drop support for Python 3.6
* Drop support for Python 3.7
* Add support for Python 3.10
* Add support for Django 4.0
* Add support for Wagtail 3.0

## [0.1.0] - 2021-07-18

* Added spanish translations
* Added social network management with wagtail site settings


src/wagtail_sb_socialnetworks/migrations/0001_initial.py
# Generated by Django 3.2.5 on 2021-07-18 02:40

import django.db.models.deletion
import modelcluster.fields
from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ("wagtailimages", "0023_add_choose_permissions"),
        ("wagtailcore", "0062_comment_models_and_pagesubscription"),
    ]

    operations = [
        migrations.CreateModel(
            name="SocialNetwork",
            fields=[
                (
                    "id",
                    models.BigAutoField(
                        auto_created=True,
                        primary_key=True,
                        serialize=False,
                        verbose_name="ID",
                    ),
                ),
                ("name", models.CharField(max_length=63, verbose_name="Name")),
                (
                    "icon",
                    models.ForeignKey(
                        on_delete=django.db.models.deletion.CASCADE,
                        to="wagtailimages.image",
                        verbose_name="Icon",
                    ),
                ),
            ],
            options={
                "verbose_name": "Social network",
                "verbose_name_plural": "Social networks",
            },
        ),
        migrations.CreateModel(
            name="SocialNetworkSettings",
            fields=[
                (
                    "id",
                    models.BigAutoField(
                        auto_created=True,
                        primary_key=True,
                        serialize=False,
                        verbose_name="ID",
                    ),
                ),
                (
                    "site",
                    models.OneToOneField(
                        editable=False,
                        on_delete=django.db.models.deletion.CASCADE,
                        to="wagtailcore.site",
                    ),
                ),
            ],
        ),
        migrations.CreateModel(
            name="SocialNetworkProfile",
            fields=[
                (
                    "id",
                    models.BigAutoField(
                        auto_created=True,
                        primary_key=True,
                        serialize=False,
                        verbose_name="ID",
                    ),
                ),
                ("profile_url", models.URLField(verbose_name="Profile URL")),
                (
                    "social_network",
                    models.ForeignKey(
                        on_delete=django.db.models.deletion.CASCADE,
                        to="wagtail_sb_socialnetworks.socialnetwork",
                        verbose_name="Social network",
                    ),
                ),
            ],
            options={
                "verbose_name": "Social network profile",
                "verbose_name_plural": "Social network profiles",
            },
        ),
        migrations.CreateModel(
            name="SiteSocialNetworkProfile",
            fields=[
                (
                    "id",
                    models.BigAutoField(
                        auto_created=True,
                        primary_key=True,
                        serialize=False,
                        verbose_name="ID",
                    ),
                ),
                (
                    "sort_order",
                    models.IntegerField(blank=True, editable=False, null=True),
                ),
                (
                    "profile",
                    models.ForeignKey(
                        on_delete=django.db.models.deletion.CASCADE,
                        related_name="+",
                        to="wagtail_sb_socialnetworks.socialnetworkprofile",
                        verbose_name="Social network profile",
                    ),
                ),
                (
                    "sitesettings",
                    modelcluster.fields.ParentalKey(
                        on_delete=django.db.models.deletion.CASCADE,
                        related_name="site_socialnetworks",
                        related_query_name="site_socialnetwork",
                        to="wagtail_sb_socialnetworks.socialnetworksettings",
                        verbose_name="Site settings",
                    ),
                ),
            ],
            options={
                "unique_together": {("sitesettings", "profile")},
            },
        ),
    ]
